File: mod/1.20.x/fabric/src/main/java/net/wiredtomato/waygl/fabric/ModMenuIntegration.java
package net.wiredtomato.waygl.fabric;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import net.wiredtomato.waygl.config.ConfigManager;

public class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ConfigManager::createConfigScreen;
    }
}

File: mod/fabric/src/main/kotlin/net/wiredtomato/waygl/ModMenuIntegration.kt
package net.wiredtomato.waygl

import com.terraformersmc.modmenu.ModMenuModMenuCompat
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import net.wiredtomato.waygl.config.Config

object ModMenuIntegration: ModMenuModMenuCompat() {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent -> Config.YACL.generateScreen(parent) }
    }
}